### ISSUE TITLE
Could not import (fastapi.)HTTPException  but starlette OK

### ISSUE THREAD

**Issue by JSchoreels** [tiangolo/uvicorn-gunicorn-fastapi-docker#4]
Hi tiangolo.

Thank you for your docker images

However, with this one "FROM tiangolo/uvicorn-gunicorn-fastapi:python3.7-alpine3.8"

I had to import HTTPException from starlette because it wasn't known in fastapi, in my import.

fixed for me but I thought it would be interesting for you to know

**Comment by AlexWootton** [comment]
Same for me. When running `tiangolo/uvicorn-gunicorn-fastapi:python3.7`, importing `HTTPException` from `fastapi` results in the following error:

```
ImportError: cannot import name 'HTTPException' from 'starlette' (/usr/local/lib/python3.7/site-packages/starlette/__init__.py)
```

To get around this I am importing directly from Starlette as @JSchoreels mentions above.

### Workaround

instead of:

```
from fastapi import HTTPException
```

use:

```
from starlette.exceptions import HTTPException
```

**Comment by tiangolo** [comment]
Thanks for the report guys. It should be fixed now.

## Details

The problem was that this image had an older version of FastAPI installed. A version from before FastAPI had its own `HTTPException`.

But the root problem is that when I released a new version of FastAPI, this image was not updated automatically.

But I fixed it here: https://github.com/tiangolo/fastapi/pull/65

Now, when there's a new version of FastAPI, a build of this image will be triggered automatically, so it will always be fresh.

Please try a recent version with:

```bash
docker pull tiangolo/uvicorn-gunicorn-fastapi:python3.7-alpine3.8
```

and

```bash
docker pull tiangolo/uvicorn-gunicorn-fastapi:python3.7
```

and let me know if it's working for you.

## `HTTPException` details:

FastAPI has its own `HTTPException` to be able to provide custom headers in it. Which is useful for some kinds of authentication/security. Apart from that, it's equivalent to Starlette's `HTTPException` (inherits from it).

In summary, you can also use Starlette's `HTTPException` directly and it should work well too.

**Comment by AlexWootton** [comment]
Perfect, thanks @tiangolo

For what it’s worth I decided it was probably better practice to use the Docker image that this image is based on and install FastAPI and any dependencies from a requirements.txt in one fell swoop!

Good to know that this image should always be running the latest release from now on though 👌

Oh and thanks for all your hard work. I’ve only been playing with FastAPI for a short while but from what I’ve seen of it so far it’s great! Excited to use it in a project in the near future ☺️

**Comment by tiangolo** [comment]
> For what it’s worth I decided it was probably better practice to use the Docker image that this image is based on and install FastAPI and any dependencies from a requirements.txt in one fell swoop!

I agree! In fact, I do that in the project generators: https://github.com/tiangolo/full-stack-fastapi-postgresql

> Good to know that this image should always be running the latest release from now on though :ok_hand: 

:ok_hand: :heavy_check_mark: 

> Oh and thanks for all your hard work. I’ve only been playing with FastAPI for a short while but from what I’ve seen of it so far it’s great! Excited to use it in a project in the near future :relaxed: 

That's great to hear! :tada: :smile:

**Comment by JSchoreels** [comment]
Thank you for the feedback @tiangolo ! Closing the issue

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
